# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I have chosen Lane 2: Refresh / Content Opportunity Scoring as my provisional lane. I selected this lane because it can help content and SEO teams identify which pages may need attention first instead of reviewing every page manually. Using the search-performance and content data available in the starter dataset, I will explore patterns that may help prioritize pages for refreshing, expanding, protecting, pruning, or monitoring. I will first use data analysis and simple rules, and later explore whether machine learning can improve the prioritization when multiple factors are involved. The main goal is not just to train a model, but to create a clear and useful ranked list that can support better decisions.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

Research Question:
Which content pages should be reviewed first based on their search performance, freshness, and engagement data?

Decision and Action:
This project will help content editors or SEO teams decide which pages need attention first instead of checking every page manually. After reviewing the ranked results, the team can decide whether a page should be refreshed, expanded, protected, pruned, or simply monitored. The results will only support the team in making a decision; they will not automatically decide what action should be taken.

Unit of Analysis:
The unit of analysis will be one anonymized content page. Each row in the starter dataset represents one content item and contains information about its content and search performance.

Expected Output:
The expected output will be a ranked list of pages that may need attention. The list may include a priority score, the reason why a page was selected, a suggested action, and a confidence level.

Cost of a Wrong Recommendation:
If the system gives high priority to a page that does not actually need attention, an editor may waste time reviewing or changing it. An unnecessary change could also affect a page that is already performing well. On the other hand, if an important page receives a low priority, the team may miss a useful opportunity to improve or review it. For this reason, the results should be used as guidance, and a person should review the page before taking any action.

Why Data or Machine Learning Can Help:
The dataset contains different types of information related to content, search performance, freshness, and engagement. A simple rule can be used as a starting point, while data analysis and machine learning may help identify patterns involving several factors at the same time. Machine learning will only be useful if it performs better than a simple and understandable rule.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [8]:
import pandas as pd

dataset_url = "https://raw.githubusercontent.com/ArshadNazir1/flyrank-ml-internship-starter/main/data/raw/content_refresh_anonymized.csv"

data = pd.read_csv(dataset_url)

pages_with_lower_impressions = (
    data["impressions_last_30d"] < data["impressions_prev_30d"]
).sum()

percentage = (pages_with_lower_impressions / len(data)) * 100

print("Total content items:", len(data))
print("Content items with lower impressions:", pages_with_lower_impressions)
print("Percentage with lower impressions:", round(percentage, 2), "%")

Total content items: 30000
Content items with lower impressions: 19716
Percentage with lower impressions: 65.72 %


The starter dataset contains 30,000 content items. Out of these, 19,716 content items received fewer impressions in the latest 30 days compared with the previous 30 days. This represents approximately 65.72% of the dataset. This observed decrease shows that there are many potential content-review candidates, which makes the selected lane worth investigating further. However, lower impressions alone do not mean that a page needs to be refreshed, because search performance may be affected by several other factors. Therefore, further analysis is needed before recommending any action.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project can identify observed patterns in content and search-performance data and use them to support decisions about which pages may need further review. The results can provide useful direction for prioritizing content, but they cannot prove why a page gained or lost impressions, clicks, or engagement. The analysis also cannot predict Google’s algorithm or guarantee that refreshing a page will improve its performance. Other factors may influence search performance, so the recommendations should be treated as decision support and reviewed by a person before making any changes.

## Self-check

Before you submit, confirm each line honestly:

- [x ] Every section above is filled — markdown thinking AND the code that backs it
- [x ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x ] No client names, URLs, or private queries anywhere
- [x ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.